antora/ec-policies-antora-extension/index.js

```diff
@@ -0,0 +1,201 @@
+            failureMsg, effectiveOn, file, row
+          })
+        }
+      }
+    })
+
+    // Group the rules by their package heading
+    return helpers.groupBy(output, a => a.pkgHeading)
+  },
+
+  processBundlesData: (rawData) => {
+    const output = {}
+    // For now we're only looking at the task-bundles since
+    // the pipeline-bundles are not used in any of the policies
+    const data = rawData["task-bundles"]
+
+    Object.keys(data).forEach(k => {
+      output[k] = data[k].map(d => {
+        const repoUrl = hbsHelpers.repoUrl(k)
+        // Todo: These are quay.io specific and should not be
+        const digestUrl = `${repoUrl}/manifest/${d.digest}`
+        const tagUrl = `$(repoUrl)?tab=tags&tag=${d.tag}`
+        const shortDigest = d.digest.split(":")[1].slice(0,12)
+
+        return {
+          ...d,
+          digestUrl,
+          tagUrl,
+          shortDigest,
+        }
+      })
+    })
+
+    return output
+  },
+
+  // Convert modules/ROOT/templates/foo.hbs to modules/ROOT/pages/foo.adoc
+  templateToPage: (path) => {
+    return path.replace(RegExp('/templates/'),'/pages/').replace(RegExp('.hbs$'), '.adoc')
+  },
+
+  // Prepare data for adding a dynamic page to the pages list
+  prepDynamicPage: (fileSrc, pageContent) => {
+    const path = helpers.templateToPage(fileSrc.path)
+    const basename = helpers.templateToPage(fileSrc.basename)
+    const contents = Buffer.from(pageContent)
+    const stem = fileSrc.stem
+    return {
+      contents,
+      path,
+      src: {
+        path,
+        basename,
+        stem,
+        extname: '.adoc',
+        origin: 'template', // Not sure about this...
+      }
+    }
+  },
+
+}
+
+module.exports.register = function() {
+  this.on('contentAggregated', ({ contentAggregate }) => {
+    // Skip modules other than the the ec-policies module
+    contentAggregate.filter(c => c.name == 'ec-policies').forEach(content => {
+      // Import yaml
+      const yaml = this.require('js-yaml')
+
+      // Find and load the two data files
+      const rawAnnotationsData = helpers.dataFromAntoraFile(yaml, content, 'rule_annotations.json')
+      const rawBundlesData = helpers.dataFromAntoraFile(yaml, content, 'acceptable_tekton_bundles.yml')
+
+      // Massage the data so the templates can be clean and tidy
+      const pipelineAnnotations = helpers.processAnnotationsData(rawAnnotationsData, "data.policy.pipeline")
+      const releaseAnnotations = helpers.processAnnotationsData(rawAnnotationsData, "data.policy.release")
+      const acceptableBundles = helpers.processBundlesData(rawBundlesData)
+
+      // Import Handlebars and register helpers and partials
+      const Handlebars = this.require('handlebars')
+      Object.keys(hbsHelpers).forEach((k) => {
+        Handlebars.registerHelper(k, hbsHelpers[k])
+      })
+
+      // Templates are in the templates directory
+      // If they start with '_' they are treated as partials instead of pages
```

Comment:
Why not have them in a "partials" directory?

Reply:
:shrug:

Reply:
It's a nod to ruby on rails, but I suppose it removes the need for an explanation...